### PR TITLE
Added enhanced nav functionality for browsers which cut the mustard

### PR DESCRIPTION
**Pull request by CSergienko**
Added some JS magic to AJAX in the menu html fragment. Some basic animation and layout stuff as well.

### CODE REVIEW

File: fed/src/js/ui/menu.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+        FancyNavToggle.addEventListener('click', function(e) {
+            e.preventDefault();
+
+            FancyNav.classList.toggle('is-open');
+        });
+
+        MenuFragment = FancyNav.children[1];
+
+        console.log(MenuFragment);
```
Comment (markupguy) [inline fed/src/js/ui/menu.js]: You can probably remove this when you're happy

File: fed/src/sass/components.enhanced/ui/_organism-fancynav.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    visibility: hidden;
+    z-index: -100;
+
+    &.is-open {
+        transform: translateX(0);
+        transform: translate3d(0, 0, 0);
+        opacity: 1;
+        visibility: visible;
+        z-index: 300;
+        overflow-y: scroll;;
```
Comment (markupguy) [inline fed/src/sass/components.enhanced/ui/_organism-fancynav.scss]: Trivial but double ;;

File: fed/src/sass/components.enhanced/ui/_organism-fancynav.scss [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    &-toggle {
+        position: absolute;
+        right: 0;
+        width: scale(delta);
+        height: scale(delta);
```
Comment (markupguy) [inline fed/src/sass/components.enhanced/ui/_organism-fancynav.scss:32]: Never seen delta before, out of interest, mind levelling me up?

File: nurseconnect/templates/core/header.html [hunk 4]
```diff
@@ -13,7 +13,7 @@
             <nav class="NavBar NavBar--siteHead" role="menubar">
                 <ul class="NavBar-navGroup" role="menu">
                     <li class="NavBar-navItem{% if active == "menu" %} NavBar-navItem--active{% endif %}"
-                        role="presentation">
+                        role="presentation" data-enhanced-nav>
```
Comment (markupguy) [inline nurseconnect/templates/core/header.html:16]: is role="presentation" necessary here?

**Review by markupguy — COMMENTED** [review]
Very nice @CSergienko nothing huge here. +1

File: fed/src/sass/components.enhanced/ui/_organism-fancynav.scss (hunk 3 above)
Comment (CSergienko) [reply fed/src/sass/components.enhanced/ui/_organism-fancynav.scss:32]: Basically, in variables.scss I have a sass map callsed `$scale` with the various size values used throughout the site for type and spacing, and I wrote this function to grab it and either return a rem/em value, or fallback to a px value for the basic stylesheets.

```
@function scale($key, $unit: "rem") {
    $scale: map-get($scale, $key);

    @if $enhanced == true {
        @if $unit == "rem" {
            @return rem($scale);
        } @elseif $unit == "em" {
            @return em($scale);
        }
    } @else {
        @return unquote("#{$scale}px");
    }
}
```

File: nurseconnect/templates/core/header.html (hunk 4 above)
Comment (CSergienko) [reply nurseconnect/templates/core/header.html:16]: I found this looking for WAI-ARIA compliant navigation patterns, and this was a recommended role to be assigned to the containing `<li>`.

Comment (markupguy) [reply nurseconnect/templates/core/header.html:16]: Ah, because it hooks to a menu? Since the others are menuitem. Makes sense I guess.

**Review by markupguy — COMMENTED** [review]
+1

**Pull request merged**